src/partition/ReceivedPartition.cpp

```diff
@@ -119,21 +119,25 @@ void ReceivedPartition::compute()
   PRECICE_DEBUG("Tag vertices for filtering: 1st round.");
   _mesh->computeState(); // normals need to be ready for NP mapping
   // go to both meshes, vertex is tagged if already one mesh tags him
-  if (_fromMapping)
-    _fromMapping->tagMeshFirstRound();
-  if (_toMapping)
-    _toMapping->tagMeshFirstRound();
+  for (mapping::PtrMapping fromMapping : _fromMappings) {
+    fromMapping->tagMeshFirstRound();
+  }
+  for (mapping::PtrMapping toMapping : _toMappings) {
+    toMapping->tagMeshFirstRound();
+  }
```

Comment:
We could refactor this into a separate function.
This would make this long function a bit more readable.

```suggestion
  tagMeshFirstRounds();
```